File: src/main/resources/js/model/address.js
class Address {
    constructor(street = "", postal = "", city = "", country = "") {
        this.street = street;
        this.postal = postal;
        this.city = city;
        this.country = country;
    }

    loadJson(json) {
        this.street = json.street;
        this.postal = json.postal;
        this.city = json.city;
        this.country = json.country;
    }

    getText() {
        let result = (this.postal + " " + this.city).trim();
        if (this.country.trim() !== "") {
            result = (result + " (" + this.country + ")").trim();
        }

        if (this.street !== "") {
            if (result === "") result = this.street;
            else result = this.street + ", " + result;
        }

        return result;
    }
}

File: src/main/java/be/vsol/util/Address.java
package be.vsol.util;

public class Address {

    public static String format(String street, String postal, String city, String country) {
        String result = (postal + " " + city).trim();
        if (!country.isBlank()) {
            result = (result + " (" + country + ")").trim();
        }

        if (!street.isBlank()) {
            if (result.isBlank()) result = street;
            else result = street + ", " + result;
        }

        return result;
    }

}
